src/table.js

```diff
@@ -927,45 +932,75 @@ Table.prototype.insert = function(entries, callback) {
 Table.prototype.mutate = function(entries, callback) {
   entries = flatten(arrify(entries));
 
-  var grpcOpts = {
-    service: 'Bigtable',
-    method: 'mutateRows',
-  };
+  var requestsMade = 0;
 
-  var reqOpts = {
-    objectMode: true,
-    tableName: this.id,
-    entries: entries.map(Mutation.parse),
-  };
-
-  var mutationErrors = [];
-
-  this.requestStream(grpcOpts, reqOpts)
-    .on('error', callback)
-    .on('data', function(obj) {
-      obj.entries.forEach(function(entry) {
-        // Mutation was successful.
-        if (entry.status.code === 0) {
-          return;
-        }
+  var maxRetries = typeof this.maxRetries === 'number' ? this.maxRetries : 3;
+  var pendingEntryIndices = new Set(entries.map((entry, index) => index));
+  var entryToIndex = new Map(entries.map((entry, index) => [entry, index]));
+  var mutationErrorsByEntryIndex = new Map();
 
-        var status = commonGrpc.Service.decorateStatus_(entry.status);
-        status.entry = entries[entry.index];
-
-        mutationErrors.push(status);
+  var onBatchResponse = error => {
+    if (pendingEntryIndices.size !== 0 && requestsMade <= maxRetries) {
+      setTimeout(
+        makeNextRequestBatch,
+        retryRequest.getNextRetryDelay(requestsMade)
```

Comment:
@kolodny isn't this already done for us in retry-request? https://github.com/stephenplusplus/retry-request/blob/1b666a7d9b75599a99e84d78ee96fa4376e53c7e/index.js#L155

Reply:
This is for followup requests due to application errors (a mutation failed). With `retry-request` the request is only retried when the status code matches some value, in this case, the code is always 200 and the error happens way later in a future stream emit event. Retry request isn't the one retrying the request at all. The snippet linked is only happening in `retryAfterDelay` which we would never hit.

Technically the setTimeout here can live in retry-request and if the `currentRetryAttempt` is `> 0` do the initial request after whatever timeout is appropriate. However I feel it's more correct for the timeout to live here since the library can choose to make it a leading request (i.e. do the followup request without delay)

Reply:
Sorry, I thought the initial holding logic was in retry-request. I mentioned that in that email thread we had, and thought it was implemented. Sorry for the confusion. I think that's the right way to go-- if retry-request is given `options.currentRetryAttempt`, it should hold until `getNextRetryDelay(options.currentRetryAttempt)`. This means the user-land application doesn't need to know about `getNextRetryDelay` (it is not intended to be a public method), and they can let retry-request worry about calculating those delays. If it ever creates problems in the future, we can always expose a setting in retry-request to disable the initial delay.

Reply:
Sounds good. I sent stephenplusplus/retry-request#19 for review, when that gets published, common-grpc will need to push a release to use that updated module, then these two lines can change to `makeNextRequestBatch()`

Reply:
done